### PR TITLE
feat: implement migrationNameTimestampFormat config option

### LINKED ISSUE #97: Custom timestamp format for files generated by command "new"
Just thinking out loud: since this package is based on migrate-mongo (https://github.com/seppevs/migrate-mongo/) I think it's nice to have transition plan to how to switch from migrate-mongo JS to mongo-migrate-ts.

Anyway, for this particular case, during that switch (from migrate-mongo JS to mongo-migrate-ts) I faced the issue of different format of files:

image

- `1697236860399_Migration.ts` | generated by mongo-migrate-ts
- `20230105075728-migration.js` | generated by migrate-mongo JS

So this is enhancement suggestion: unify format for file naming OR add the config option (`migrations.json`) to use specific time format.

📓 The difference of timestamps:
- https://github.com/mycodeself/mongo-migrate-ts/blob/ba6684baa78db17d1241e47848955507c1dcae38/lib/commands/new.ts#L48
- https://github.com/seppevs/migrate-mongo/blob/3753b2bc52436d06482efc8442d10e93960b83f8/lib/actions/create.js#L24

### PR DESCRIPTION
**Pull request by Mifrill**
closes #97

### CODE REVIEW

File: __tests__/config.test.ts [hunk 1]
```diff
@@ -45,13 +45,13 @@ describe('config', () => {
     const processedConfig = processConfig({
       ...configMock,
       uri,
-      database: undefined
+      database: undefined,
```
Comment (Mifrill) [inline __tests__/config.test.ts:48]: 📓 automatic prettier changes

**Comment by Mifrill** [comment]
📝 TODO: update README accordingly

**Comment by Mifrill** [comment]
📝 TODO: fix `yarn audit --level high --no-progress --frozen-lockfile`: https://app.circleci.com/jobs/github/mycodeself/mongo-migrate-ts/593

**Comment by Mifrill** [comment]
📓 [The failed build](https://app.circleci.com/jobs/github/mycodeself/mongo-migrate-ts/603) is not related to those changes.

**Comment by Mifrill** [comment]
📝 TODO: fix time freeze issue in tests.

**Comment by Mifrill** [comment]
Hi @mycodeself please take a look at this implemention 🙏

**Comment by Mifrill** [comment]
Hey @mycodeself just a gentle reminder, please check when you get a chance 🙏

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mongo-migrate-ts",
-  "version": "1.0.13",
+  "version": "1.1.0",
```
Comment (mycodeself) [inline package.json]: Do not update the version, semantic-release will do it automatically

File: lib/utils/timestamp.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      this.date.getUTCFullYear(),
+      this.date.getUTCMonth(),
+      this.date.getUTCDate(),
+      this.date.getUTCHours(),
+      this.date.getUTCMinutes(),
+      this.date.getUTCSeconds(),
+      this.date.getUTCMilliseconds()
+    );
+  }
+}
```
Comment (mycodeself) [inline lib/utils/timestamp.ts]: would it be possible not to use a class here and just use a function?

File: lib/commands/new.ts [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
 import { TemplateFileNotFoundError } from '../errors';
+import { Timestamp } from '../utils/timestamp';
 
 interface CommandNewOptions {
   migrationsDir: string;
-  migrationName?: string;
+  migrationName: {
+    title?: string;
+    timestampFormat: string;
+  };
```
Comment (mycodeself) [inline lib/commands/new.ts]: I think it would be simpler to keep these parameters separate

**Review by mycodeself — CHANGES_REQUESTED** [review]
hey @Mifrill 
thanks for your contribution and sorry for the slow response, I've suggest few changes, can you check please?

File: package.json (hunk 2 above)
Comment (Mifrill) [reply package.json]: makes sense 👍, adjusted: https://github.com/mycodeself/mongo-migrate-ts/pull/99/commits/5ec0cd87fc697e37a2e00205a757b715121c1386

File: lib/utils/timestamp.ts (hunk 3 above)
Comment (Mifrill) [reply lib/utils/timestamp.ts]: no problem, edited: https://github.com/mycodeself/mongo-migrate-ts/pull/99/commits/0dc0f02bc4e8c105950d3598a5f081e2313b68d5

File: lib/commands/new.ts (hunk 4 above)
Comment (Mifrill) [reply lib/commands/new.ts]: okay 👍. Separated: https://github.com/mycodeself/mongo-migrate-ts/pull/99/commits/181d1e519d3f69bd7608840cc5ad7c35557e24c8

**Comment by Mifrill** [comment]
Hey @mycodeself thanks for the review 👍. Please check the updated version.

**Comment by Mifrill** [comment]
Hey @mycodeself just a reminder, please check the updated version when you get a time 🙏

**Review by mycodeself — APPROVED** [review]

**Comment by mycodeself** [comment]
@Mifrill sorry for the delay, I've just merged your PR, and must be available in the next release.

Thanks a lot for your contribution!

**Pull request merged**